tests/utilities/test_app.py

```diff
@@ -177,3 +222,92 @@ def disable_input(self) -> None:
+) -> ContextManager[MockedTimeMoveClockForward]:
+    single_tick_duration = 1.0 / ticks_granularity_fps
+
+    pending_sleep_events: list[tuple[float, asyncio.Event]] = []
+
+    @contextlib.contextmanager
+    def mock_textual_timers_context_manager():
+        # N.B. `start_time` is not used, but it is useful to have when we set breakpoints there :-)
+        start_time = current_time = monotonic()
+
+        # Our replacement for "textual._timer.Timer._sleep":
+        async def sleep_mock(duration: float) -> None:
+            event = asyncio.Event()
+            target_event_monotonic_time = current_time + duration
+            pending_sleep_events.append((target_event_monotonic_time, event))
+            # Ok, let's wait for this Event
+            # (which can only be "unlocked" by calls to `move_clock_forward()`)
+            await event.wait()
+
+        # Our replacement for "textual._timer.Timer.get_time" and "textual.message.Message._get_time":
+        def get_time_mock() -> float:
+            nonlocal current_time
+
+            # let's make the time advance slightly between 2 consecutive calls of this function,
+            # within the same order of magnitude than 2 consecutive calls to ` timer.monotonic()`:
+            current_time += 1.1e-06
+
+            return current_time
+
+        async def move_clock_forward(*, seconds: float) -> tuple[float, int]:
+            nonlocal current_time, start_time
+
+            ticks_count = ceil(seconds * ticks_granularity_fps)
+            activated_timers_count_total = 0
+            for tick_counter in range(ticks_count):
+                current_time += single_tick_duration
+                activated_timers_count = check_sleep_timers_to_activate()
+                activated_timers_count_total += activated_timers_count
+                # Let's give an opportunity to asyncio-related stuff to happen,
+                # now that we likely unlocked some occurrences of `await sleep(duration)`:
+                if activated_timers_count:
+                    await let_asyncio_process_some_events()
+
+            await let_asyncio_process_some_events()
+
+            return current_time, activated_timers_count_total
+
+        def check_sleep_timers_to_activate() -> int:
+            nonlocal pending_sleep_events
+
+            activated_timers_count = 0
+            for i, (target_event_monotonic_time, event) in enumerate(
+                pending_sleep_events
+            ):
+                if current_time < target_event_monotonic_time:
+                    continue  # not time for you yet, dear awaiter...
+                # Right, let's release this waiting event!
+                event.set()
+                activated_timers_count += 1
+                # ...and remove it from our pending sleep events list:
+                del pending_sleep_events[i]
+
+            return activated_timers_count
+
+        with mock.patch("textual._timer._TIMERS_CAN_SKIP", new=False), mock.patch(
```

Comment:
we end up with 3 things to mock in our tests to simulate time in Textual; not too bad! :slightly_smiling_face: